### PR TITLE
Adds NeoCSV as a dependency and provides  asCSV and asJSON for ...

### PR DESCRIPTION
**Pull request by tblanchard**
MySQLRow.  Also nicer column indexing allowing int or name.

Basically a few methods I added after finding it nearly impossible to generate a report using the result processing methods available.

### CODE REVIEW

**Comment by tblanchard** [comment]
I should add some tests before we merge this.  Also, tests are failing due to permissions. 

@gcotelli can you let me know how we fix this?  Also would like a second  set of eyes on this change.

**Comment by gcotelli** [comment]
I will take a look at the permissions issue.

**Comment by gcotelli** [comment]
@tblanchard the build is running again. Take a look because the baseline is referencing a non-existent NeoCSV group or package.

```smalltalk
MetacelloNameNotDefinedError: project group, or package named: 'NeoCSV-CORE' not found when used in requires: or includes: field of package: 'MySQL-Core' for version: baseline of BaselineOfMySQL.
```

**Comment by tblanchard** [comment]
@gcotelli Thanks, I've pushed a fix for that - mistyped package name.

**Comment by gcotelli** [comment]
~~Looks like we hit a smalltalkCI bug because it is trying to write the coverage info into an invalid place:~~
```
Writing LCOV coverage info to: /home/runner/work/Pharo-MySQL/Pharo-MySQL/coverage/lcov.info
FileDoesNotExistException: '/home/runner/work/Pharo-MySQL/Pharo-MySQL/MySQL-Core/MySQLBindError.class.st'
```

~~Maybe it's because the project has the package folders in the root one. I always put my packages inside a source folder and never hitted this error.~~

**Comment by gcotelli** [comment]
@tblanchard I think I've found the problem. Some of the files have an invalid case, for example `MySQLResultSet` was written somehow as `MysqlResultSet.class.st` and smalltalkCI needs to locate the file to calculate the hash to export the coverage data. I will try to locate the invalid files and rename them but give me some time.

**Comment by gcotelli** [comment]
Ok, now that the build is working let me see the changes.

I think that the CSV conversion is better handled by adding the methods as extensions into another package and providing an optional group to load in the baseline. Because right now it adds a hard dependency to NeoCSV for everyone wanting to use the package, even if they don't want to use the CSV conversion stuff. What do you think?

**Comment by tblanchard** [comment]
@gcotelli I kind of agree with you on the CSV, I was trying to find a nice place to put it but maybe the best place is my own application code.   I am putting together a BaselineOf for a project I am starting and just trying to get all the dependencies to load cleanly.  I think I will revert the CSV piece and add some tests before merging.  Thanks for your input and for quickly diagnosing the issue!

**Comment by tblanchard** [comment]
Oh, actually, I see what you mean about making it an optional load from MySQL.  I will look at that.

**Comment by astares** [comment]
@tblanchard / @gcotelli 

I would also prefer to not load NeoCSV with the core group. The **Core** should really only be the essential stuff and not anyone needs or uses the csv additions.

Why not have an additional group "Core-extended" (that depends on "Core) in baseline with the additional dependency on NeoCSV -  so one can pick either "Core" group (for rare stuff) or the "Core-extended" for Core + additional support like CSV.

**Comment by astares** [comment]
Only a minor thing: maybe you should call own groups "Core" instead "core", "Glorp" instead of "glorp", .... 
Then it is like in other BaselineOf classes.

Usually only meta-groups like "all" or "default" are lowercase.

**Comment by tblanchard** [comment]
> Only a minor thing: maybe you should call own groups "Core" instead "core", "Glorp" instead of "glorp", ....
> Then it is like in other BaselineOf classes.
> 
> Usually only meta-groups like "all" or "default" are lowercase.

I moved it out to a group csv which requires core.  Do you think we will break any existing projects by renaming?

**Comment by gcotelli** [comment]
I also use group names capitalized. 
But yes, if you remove the old group names this will break dependencies for sure. Maybe we can add the new groups capitalized and a capitalized version of the previous groups (so `core` will continue to work, but also `Core`). And give a notice on the release notes of this version and a time window for users to migrate. Once the time window has passed we can delete the lowercase groups in the next release.

**Comment by tblanchard** [comment]
@gcotelli That sounds like the best idea.

File: BaselineOfMySQL/BaselineOfMySQL.class.st [hunk 1]
```diff
@@ -54,6 +58,10 @@ BaselineOfMySQL >> setUpPackages: spec [
 	spec
 		package: 'MySQL-Core' with: [ spec requires: #('NeoJSON-Core' 'ZTimestamp') ];
 		group: 'core' with: 'MySQL-Core'.
```
Comment (gcotelli) [inline BaselineOfMySQL/BaselineOfMySQL.class.st]: Add a new `Core` group

File: BaselineOfMySQL/BaselineOfMySQL.class.st [hunk 2]
```diff
@@ -54,6 +58,10 @@ BaselineOfMySQL >> setUpPackages: spec [
...
+		
+	spec
+		package: 'MySQL-CSV' with: [ spec requires: #('MySQL-Core' 'Neo-CSV-Core') ];
+		group: 'csv' with: 'MySQL-CSV'.
```
Comment (gcotelli) [inline BaselineOfMySQL/BaselineOfMySQL.class.st]:
```suggestion
		group: 'CSV' with: 'MySQL-CSV'.
```

**Review by gcotelli — CHANGES_REQUESTED** [review]
- Please update the group names as discussed
- The new functionality is missing some test cases

**Comment by gcotelli** [comment]
@tblanchard apparently the problem is that you're trying to define a group Glorp and it collides with the Glorp baseline definition. Try a different name for the group? Sometimes Metacello is pretty awkward with the errors.

File: BaselineOfMySQL/BaselineOfMySQL.class.st (hunk 1 above)
Comment (tblanchard) [reply BaselineOfMySQL/BaselineOfMySQL.class.st]: core is now Core

**Comment by tblanchard** [comment]
I'm done screwing with this.  I have the methods I need to support the asCSV and I will add that in my application.  I did make the changes.

> Only a minor thing: maybe you should call own groups "Core" instead "core", "Glorp" instead of "glorp", ....
> Then it is like in other BaselineOf classes.
> 
> Usually only meta-groups like "all" or "default" are lowercase.

glorp is staying glorp because Glorp breaks the tests as it conflicts with package Glorp and I'm out of patience with this.  I have a project to kick off.  Happy to review someone else's efforts towards this.  I've also given up on putting in CSV at all.

**Pull request merged**